lib/logstash/plugin_mixins/event_support/event_factory_adapter/fallback_impl.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require 'jruby'
+
+module LogStash::PluginMixins::EventSupport::EventFactoryAdapter::FallbackImpl
+
+  # The event_factory method is effectively final and should not be re-defined by plugins.
+  #
+  # @return an event factory object with a `new_event(Hash)` API
+  def event_factory
+    @_event_factory || JRuby.reference(self).synchronized do
+      @_event_factory ||= create_event_factory
+    end
```

Comment:
I _think_ that we can afford to be simpler here in our fallback implementation and just return the `BasicEventFactory` singleton. By definition, if we are using the fallback implementation we aren't able to use core features so it doesn't make sense for core to need to hook into this.

```suggestion
    BasicEventFactory::INSTANCE
```

Reply:
okay I did a slightly different variation: `@_event_factory ||= BasicEventFactory::INSTANCE`
which looks weird but still communicates an intent of a factory being lazy initialized (as I think we'll revisit some stateful form of `create_event_factory` in the future). 

on a related note I would also remove `create_event_factory` from core for now, wdyt?

the reasoning being I am not happy with doing a `create_event_factory(config = self.config)` to make it more useful for now due the fact that it treats `config['target']` in a special way ... think we should wait for the use-case when a plugin would need to instantiate an event factory manually.

Reply:
Similar topic, but from the LS core perspective

> It's private, so not necessarily something we need to align on. It is also a good hook point for future efforts in LS core that don't need backport support (things like alternate Event implementation experiments like copy-on-write or field reference parser changes), so I expect even if we remove it now we will add it back in short order.

I don't see a world in which plugins ever need to reach in under the hood and change the implementation details of the event factory. This support gem is in place so that plugins can _use_ the event factory APIs on older logstashes where they aren't provided by LS core. I see a future LS core providing alternate implementations based on a pipeline's configuration (such as copy-on-write events, events that use an alternate field reference parser, other things), but which implementation it provides is outside the scope of this support adapter gem.

So in short: LS core was merged with a [_private_ `LogStash::Plugins::EventFactorySupport#create_event_factory`](https://github.com/elastic/logstash/pull/13017/files#diff-b85d5060885a699db3d00ac4d5c77e33edeecf9540aa1f267f2a467f68fa8afeR30), but that is an implementation detail of LS core and there is no need to mirror this in the support adapter's implementation.